types/express-formidable/index.d.ts

```diff
@@ -1,35 +1,40 @@
-// Type definitions for express-formidable 1.0
+// Type definitions for express-formidable 1.2
 // Project: https://github.com/noraesae/express-formidable
 // Definitions by: Torkild Dyvik Olsen <https://github.com/tdolsen>, Evan Shortiss <https://github.com/evanshortiss>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
 import * as express from "express";
-import { Fields, Files } from "formidable";
+import { Fields, Files, EventNames } from "formidable";
 
 // Extend the express request object with attached formidable files and fields
 declare global {
   namespace Express {
     interface Request {
-      fields?: Fields | undefined;
-      files?: Files | undefined;
+      fields?: Fields ;
+      files?: Files ;
     }
   }
 }
 
 interface ExpressFormidableOptions {
-    encoding?: string | undefined;
-    uploadDir?: string | undefined;
-    keepExtensions?: boolean | undefined;
-    type?: "multipart" | "urlencoded" | undefined;
-    maxFileSize?: number | undefined;
-    maxFieldsSize?: number | undefined;
-    maxFields?: number | undefined;
-    hash?: boolean | "sha1" | "md5" | undefined;
-    multiples?: boolean | undefined;
+    encoding?: string ;
+    uploadDir?: string ;
+    keepExtensions?: boolean ;
+    type?: "multipart" | "urlencoded" ;
+    maxFileSize?: number ;
+    maxFieldsSize?: number ;
+    maxFields?: number ;
+    hash?: boolean | "sha1" | "md5" ;
+    multiples?: boolean ;
 }
 
-declare function ExpressFormidable(options?: ExpressFormidableOptions): express.RequestHandler;
+interface ExpressFormidableEvents {
```

Comment:
It would be nice to avoid any for the formidable params. Something like:

```ts
type FormidableFileEvent = {
  event: 'fileBegin',
  action: function (
    req: express.Request,
    res: express.Response,
    next: express.NextFunction,
    name: string,
    file: File) { /* your callback */ }
}

type FormidableFieldEvent = {
  event: 'fileBegin',
  action: function (
    req: express.Request,
    res: express.Response,
    next: express.NextFunction,
    name: string,
    value: string|string[]) { /* your callback */ }
}

type ExpressFormidableEvents = FormidableFileEvent| FormidableFieldEvent;
```

This is a less trivial effort though, and any will work for the time being.

Reply:
CC @joshuaharry